elide-datastore/elide-datastore-hibernate/src/main/java/com/yahoo/elide/core/filter/CaseAwareJPQLGenerator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.core.filter;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * JPQLPredicateGenerator that can generate both case sensitive/insensitive predicate fragments.
+ */
+public class CaseAwareJPQLGenerator implements FilterTranslator.JPQLPredicateGenerator {
+
+    private static final String COMMA = ", ";
+    private static final String FILTER_PATH_NOT_NULL = "Filtering field path cannot be empty.";
+    private static final String FILTER_ALIAS_NOT_NULL = "Filtering alias cannot be empty.";
```

Comment:
Maybe we should do `@slf4j` and add loggings to these?

Reply:
👍